server/src/daemon/service.rs

```diff
@@ -24,11 +24,33 @@ use super::{
     collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
 };
 
+#[derive(Debug)]
+pub struct ServiceManager {
+    sessions: HashMap<OwnedObjectPath, Session>,
```

Comment:
`Session` already has a `path` filed attached to it. Can't we use that when we need to access `path` and store `sessions` as a `Vec<Session>`?

Reply:
Getting a specific item from the hashmap through it path would be faster than iterating the vec i would say

Reply:
understood, thanks!